lib/rucio/core/quarantined_replica.py

```diff
@@ -180,11 +207,17 @@ def list_rses_with_quarantined_replicas(filters: Optional[dict[str, Any]] = None
 
     :returns: a list of RSEs.
     """
-    query = session.query(models.RSE.id).distinct(models.RSE.id).\
-        filter(models.QuarantinedReplica.rse_id == models.RSE.id).\
-        filter(models.RSE.deleted == false())
+    stmt = select(
+        models.RSE.id
+    ).distinct(
+    ).where(
+        and_(models.QuarantinedReplica.rse_id == models.RSE.id,
+             models.RSE.deleted == false())
+    )
 
     if filters and filters.get('vo'):
-        query = query.filter(getattr(models.RSE, 'vo') == filters.get('vo'))
+        stmt = stmt.where(
+            models.RSE.vo == filters.get('vo')
+        )
 
-    return [rse for (rse,) in query]
+    return [str(rseid) for rseid in session.execute(stmt).scalars().all()]
```

Comment:
Just saw #6930, when that PR is merged this query should be rewritten to:
`    return list(session.execute(stmt).scalars().all())`

If #6930 is merged before this PR then I'll modify it in this PR, otherwise I will create a new PR addressing this at a later date